library/common/extensions/filters/http/platform_bridge/filter.h

```diff
@@ -46,9 +46,18 @@ enum class IterationState { Ongoing, Stopped };
  * For more information on implementing platform filters, see the docs.
  */
 class PlatformBridgeFilter final : public Http::PassThroughFilter,
-                                   Logger::Loggable<Logger::Id::filter> {
+                                   public Logger::Loggable<Logger::Id::filter>,
+                                   public std::enable_shared_from_this<PlatformBridgeFilter> {
 public:
-  PlatformBridgeFilter(PlatformBridgeFilterConfigSharedPtr config);
+  PlatformBridgeFilter(PlatformBridgeFilterConfigSharedPtr config, Event::Dispatcher& dispatcher);
+
+  // Asynchronously trigger resumption of filter iteration, if applicable.
+  // This is a no-op if filter iteration is already ongoing.
+  void resumeDecoding();
+
+  // Asynchronously trigger resumption of filter iteration, if applicable.
+  // This is a no-op if filter iteration is already ongoing.
+  void resumeEncoding();
```

Comment:
should the onResume callbacks below be made private now that we have the async triggers exposed?

Reply:
It could be, though right now its visibility aligns with the other `on*` functions in the class (which perhaps could arguably all be made private).

Reply:
Actually, I'm mistaken, they're already private. Let me move it.